semgrep-core/src/parsing/Test_parsing.ml

```diff
@@ -29,10 +29,67 @@ let logger = Logging.get_logger [ __MODULE__ ]
 (* Helpers *)
 (*****************************************************************************)
 
+module SMap = Map.Make (String)
+module IMap = Map.Make (Int)
+
+(* Global store, because I don't want to pass it through a bunch of func calls.
+ *)
+let store = ref []
+
+(* This function just processes the exception backtrace to find the first instance where a function
+   that is not called "todo" was called.
+   It then records the line number and name of the function, adding it to a global store.
+*)
+let process_exn () =
+  let rec process (lines : string list) =
+    match lines with
+    | [] -> "failure"
+    | line :: rest -> (
+        (* Sue me. I don't want to write a real lexer and parser. *)
+        let tokens = Common.split " " line in
+        match tokens with
+        | "Called" :: "from" :: funcname :: _in :: _file :: _filename :: "line"
+          :: linenum :: _ ->
+            if String.ends_with ~suffix:"todo" funcname then process rest
+            else funcname ^ ":" ^ linenum
+        | _ -> process rest)
+  in
+  let res =
+    Printexc.get_backtrace () |> Common.split (Str.quote "\n") |> process
+  in
+  store := res :: !store;
+  ()
+
 let print_exn file e =
   let trace = Printexc.get_backtrace () in
+  process_exn ();
   pr2 (spf "%s: exn = %s\n%s" file (Common.exn_to_s e) trace)
 
+(* This function collects all the function name and line number pairs, and then
+   sorts them in descending order of frequency.
+   I'm using maps to do this so that I can achieve O(log n) insertion per increment.
```

Comment:
I usually just use a Hashtbl and then Common.hash_to_list |> Common.sort_by_value |> Common.take_safe 10 |> ...